.github/workflows/target_workflow.yaml

```diff
@@ -15,14 +22,49 @@ jobs:
     runs-on: ubuntu-latest
 
     steps:
+      - name: Set target service
+        id: service
+        run: echo "::set-output name=target_service::query-connector"
+
+      - name: Set version for main push
+        if: github.event_name == 'push'
+        id: version
+        run: |
+          # Check if there's an existing version file; if not, start from 1.0.0
```

Comment:
I wouldn't recommend doing things this way. If the version file doesn't exist, I would recommend failing the deployment outright.

Reason: Assume someone has deleted the version file in their commit on accident. You run this, and create a new 1.0.0 image. You're actually on 5.0.6. You've just overwritten your archived 1.0.0 image that someone in production may have been depending on.